src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -280,11 +283,23 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
   def getMethodConfiguration(workspaceNamespace: String, workspaceName: String, methodConfigurationNamespace: String, methodConfigurationName: String): PerRequestMessage =
     dataSource inTransaction { txn =>
       withWorkspace(workspaceNamespace, workspaceName, txn) { workspace =>
-        withMethodConfig(workspace, methodConfigurationNamespace, methodConfigurationName) { methodConfig =>
+        withMethodConfig(workspace, methodConfigurationNamespace, methodConfigurationName, txn) { methodConfig =>
           PerRequest.RequestComplete(methodConfig)
         }
       }
     }
+
+  def copyMethodConfiguration(workspaceNamespace: String, workspaceName: String, sourceMethodConfigName: MethodConfigurationName): PerRequestMessage =
+    dataSource inTransaction { txn =>
+      withWorkspace(sourceMethodConfigName.workspaceName.namespace, sourceMethodConfigName.workspaceName.name, txn) { srcWorkspace =>
+        methodConfigurationDAO.get(sourceMethodConfigName.workspaceName.namespace, sourceMethodConfigName.workspaceName.name, sourceMethodConfigName.methodConfigurationNamespace, sourceMethodConfigName.name, txn) match {
+          case Some(srcMethodConfig) =>
+            methodConfigurationDAO.save(workspaceNamespace, workspaceName, srcMethodConfig.copy(workspaceName = WorkspaceName(workspaceNamespace, workspaceName)), txn)
+            RequestComplete(StatusCodes.OK)
```

Comment:
I think we should return the newly created method config and also a 201 created status